source/lib/login.js

```diff
@@ -41,14 +43,28 @@ export async function performLogin({
 		return 'no-credentials'
 	}
 
-	const form = buildFormData({username, password})
-	let loginResult = null
+	let form = buildFormData({username, password})
+
 	try {
-		loginResult = await fetch(OLECARD_AUTH_URL, {
+		let {status: statusCode} = await rawFetch(OLECARD_AUTH_URL, {
 			method: 'POST',
 			body: form,
 			credentials: 'include',
 		})
+
+		if (statusCode >= 400 && statusCode < 500) {
```

Comment:
These three ifs are screaming "DRY me up" but I'm not sure of an easy way to do that in JS. Ideas?

Reply:
be explicit 😛 

sadly JS doesn't have pattern matching (yet?) so there's not much more you can do here.

Reply:
Oh let's definitely do

```js
switch(true) {
case(statusCode >= 400 && statusCode < 500):
  // ...
}
```

That's an excellent idea.

/s

Reply:
oh dear